x-team/xp-ui
# Add standalone components bundles

**Release Type:** *Breaking change*

Fixes https://x-team-internal.atlassian.net/browse/XP-3673

## Description

Some UI components rely on third-party lib, and those libs uses `window`. When attempting to use those UI comps in XP-Registration, it breaks the build, because Gatsby build uses SSR, where `window` is undefined.

By abstracting those UI components to a different output we can load using React.lazy feature.

Example:

```js
const SkillsSelector = React.lazy(() => import('xp-ui/lib/SkillsSelector'))
```

## Checklist


## Related PRs

branch | PR
------ | ------
`XP-3673-create-experience-page` | [XP Registration](https://github.com/x-team/xp-registration/pull/185)

## Steps to Test or Reproduce

Copy the lib files from this branch to xp-registration and run `npm run build`. The build should ran successfully.

## Impacted Areas in Application

New `xp-ui/lib/ComponentName.js` is generated when running `npm run build` in XP-UI

## Screenshots

> `npm run build` in **xp-registration**

Before:

![Screenshot 2020-04-30 at 10 48 25](https://user-images.githubusercontent.com/131859/80723118-98d5bc00-8ad6-11ea-8277-602746832773.png)

After:

<img width="656" alt="Screen Shot 2020-04-30 at 11 35 03" src="https://user-images.githubusercontent.com/131859/80723162-a68b4180-8ad6-11ea-9c33-f1d64e96a345.png">
